k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/artifactSupplier/GitArtifactSupplier.kt

```diff
@@ -0,0 +1,96 @@
+
+    override val supportedArtifact = SupportedArtifact("git", GitRepoArtifact::class.java)
+    override val supportedSortingStrategy =
+        SupportedSortingStrategy("git", GitRepoVersionSortingStrategy::class.java)
+
+    override fun getArtifactByVersion(artifact: DeliveryArtifact, version: String): PublishedArtifact? {
+        return runBlocking {
+            if (artifact is GitRepoArtifact) {
+                val versionFromIgor = igorArtifactService.getGitVersion(artifact.gitType, artifact.project, artifact.repoName, version )
+                return@runBlocking versionFromIgor.toPublishedArtifact(artifact.reference)
+            }
+            null
+        }
+    }
+
+    override fun getLatestArtifact(deliveryConfig: DeliveryConfig, artifact: DeliveryArtifact): PublishedArtifact? {
+        return runBlocking {
+            findArtifactVersions(artifact).sortedWith(artifact.sortingStrategy.comparator).firstOrNull()
+        }
+    }
+
+    override fun getLatestArtifacts(
+        deliveryConfig: DeliveryConfig,
+        artifact: DeliveryArtifact,
+        limit: Int
+    ): List<PublishedArtifact> {
+        return findArtifactVersions(artifact).sortedWith(artifact.sortingStrategy.comparator).take(limit)
+    }
+
+    override fun shouldProcessArtifact(artifact: PublishedArtifact): Boolean {
+        return artifact.version.isNotBlank()
+                && artifact.type == GIT
+                && artifact.gitMetadata?.repo?.link?.isNotBlank() ?: false
+    }
+
+    private fun findArtifactVersions(
+        artifact: DeliveryArtifact
+    ): Set<PublishedArtifact> =
+        runBlocking {
+            if (artifact is GitRepoArtifact) {
+                val versionsFromIgor =
+                    igorArtifactService.getGitVersions(artifact.gitType, artifact.project, artifact.repoName)
+                return@runBlocking versionsFromIgor.map {
+                    it.toPublishedArtifact(artifact.reference)
+                }.toSet()
+            }
+            return@runBlocking emptySet()
+        }
+
+    override suspend fun getArtifactMetadata(artifact: PublishedArtifact): ArtifactMetadata {
+        return ArtifactMetadata(
+            null,
+            artifact.gitMetadata ?: generateGitMetadata(artifact.metadata)
+        )
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun generateGitMetadata(metadata: Map<String,Any?>): GitMetadata {
+        val metadataMap = metadata as Map<String, String>
+        return GitMetadata(
+            metadataMap.getOrDefault("commitId", ""),
+            metadataMap["author"],
+            metadataMap["project"],
+            metadataMap["branch"],
+            Repo(
+                metadataMap["name"],
+                metadataMap["repoUrl"]
+            ),
+            null,
+            Commit(
+                metadataMap["commitId"],
+                metadataMap["url"],
+                metadataMap["message"]
+            )
+        )
+    }
```

Comment:
this is great. well done. for docker images, pulling git info was a pain. for git artifacts it is easier and I am glad we are managing it here.